integration-tests/tests/src/node/sync-proxy.ts

```diff
@@ -0,0 +1,114 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// NOTE: This file is only supposed to be imported from a Node.js environment
+
+import http from "node:http";
+import net from "node:net";
+import { URL } from "node:url";
+
+import { expect } from "chai";
+import Realm, { ProxyType } from "realm";
+import { importAppBefore } from "../hooks";
+import { buildAppConfig } from "../utils/build-app-config";
+import { generatePartition } from "../utils/generators";
+
+function getSyncConfiguration(user: Realm.User, partition: string, addProxyConfig: boolean): Realm.Configuration {
+  const realmConfig: Realm.Configuration = {
+    schema: [
+      {
+        name: "Dog",
+        primaryKey: "_id",
+        properties: {
+          _id: "objectId",
+          breed: "string?",
+          name: "string",
+          partition: { type: "string", default: partition },
+        },
+      },
+    ],
+    sync: {
+      user: user,
+      partitionValue: partition,
+    },
+  };
+
+  if (addProxyConfig) {
+    realmConfig.sync.proxyConfig = {
+      address: "127.0.0.1",
+      port: 9876,
+      type: ProxyType.HTTP,
+    };
+  }
```

Comment:
An alternative (but check if TS and linters are happy):
```suggestion
      proxyConfig: addProxyConfig ? {
        address: "127.0.0.1",
        port: 9876,
        type: ProxyType.HTTP,
      } : undefined,
    },
  };
```

Reply:
I can do that - with a little formatting.